proposals/bip-002.md

```diff
@@ -0,0 +1,99 @@
+# Explicit Annotated status and completeness information
+
+| Field | Value |
+| --- | --- |
+| BIP | 2 |
+| Title | Explicit Annotated status and completeness information |
+| Author | Juanjo Álvarez |
+| Status | Accepted |
+| Created | 2017-06-01 |
+| Updated | 2017-08-02 |
+
+## Abstract
+
+This BIP proposed two new official `Incomplete` and `Unannotated` roles would mark
+if the node has been annotated in any way and the quality of the current
+annotation.
+
+## Rationale
+
+At the current stage of the Babelfish project, we strive for genericity in the list
+of UAST roles and not adding very language-specific features into it until we can
+have a wider view about how to integrate or compose these roles for covering as
+much ground as possible. 
+
+This is a worthy objective but has the problem of leaving the nodes related with
+these features unannotated or with annotations that don't fully express all
+the semantic information that the language source code or the native AST have.
+
+This BIP proposes a mechanism to mark nodes with two possible completion status
+roles. This will make the UAST user have a wider view of the completeness of the
+information provided by it on specific nodes and also could enable new uses cases
+(whose specific definition is outside of the scope of this BIP) like helping debug
+drivers and provide a completeness summary, with the added benefit of providing a
+numerically observable measure of the completeness of every developed driver.
+
+## Example
+
+If a version of the UAST specification didn't have a role that could be matched
+for a "coroutine" type function supported by some language, the UAST could still
+mark those nodes as "FunctionDef", but without additional information, this would
+be providing incomplete and even misleading information to UAST users that could
+treat coroutines differently (for example when making stack call deep analysis).
+
+With this BIP implemented, the driver's programmer can add the role `Incomplete`
+to these coroutine nodes and the UAST user for that language could then look at
+the native AST properties to complete their semantic understanding of the node.
+
+## Specification
+
+### UAST Roles
+
+This change will add two new roles the UAST definition:
+
+- `Incomplete`: For nodes that have been annotated annotations doesn't express the
```

Comment:
* s/doesn't/don't/
* is some word missing between _annotated_ and _annotations_?